commands/migratecmd.go

```diff
@@ -0,0 +1,205 @@
+// inventory object by querying the inventory object in the cluster,
+// or an error if one occurred.
+func (mo *MigrateOptions) retrieveInvObjs(invInfo *resource.Info) ([]object.ObjMetadata, error) {
+	cmInvClient, err := mo.cmProvider.InventoryClient()
+	if err != nil {
+		return nil, err
+	}
+	cmObjs, err := cmInvClient.GetClusterObjs(invInfo)
+	if err != nil {
+		return nil, err
+	}
+	fmt.Fprintf(mo.ioStreams.Out, "success (%d inventory objects)\n", len(cmObjs))
+	return cmObjs, nil
+}
+
+// migrateObjs stores the passed objects in the ResourceGroup inventory
+// object and applies the inventory object to the cluster. Returns
+// an error if one occurred.
+func (mo *MigrateOptions) migrateObjs(cmObjs []object.ObjMetadata, reader io.Reader, args []string) error {
+	fmt.Fprintf(mo.ioStreams.Out, "migrate inventory to ResourceGroup...")
+	rgReader, err := mo.rgProvider.ManifestReader(reader, args)
+	if err != nil {
+		return err
+	}
+	infos, err := rgReader.Read()
+	if err != nil {
+		return err
+	}
+	// Filter the ConfigMap inventory object.
+	rgInv, err := findResourceGroupInv(infos)
+	if err != nil {
+		return err
+	}
+	rgInvClient, err := mo.rgProvider.InventoryClient()
+	if err != nil {
+		return err
+	}
+	_, err = rgInvClient.Merge(rgInv, cmObjs)
+	if err != nil {
+		return err
+	}
+	fmt.Fprintf(mo.ioStreams.Out, "success\n")
+	return nil
+}
+
+// deleteConfigMapInv removes the passed inventory object from the
+// cluster. Returns an error if one occurred.
+func (mo *MigrateOptions) deleteConfigMapInv(invInfo *resource.Info) error {
+	fmt.Fprintf(mo.ioStreams.Out, "deleting old ConfigMap inventory object...")
+	cmInvClient, err := mo.cmProvider.InventoryClient()
+	if err != nil {
+		return err
+	}
+	if err = cmInvClient.DeleteInventoryObj(invInfo); err != nil {
+		return err
+	}
+	fmt.Fprintf(mo.ioStreams.Out, "success\n")
+	return nil
+}
+
+// findResourceGroupInv returns the ResourceGroup inventory object from the
+// passed slice of objects, or nil and and error if there is a problem.
+func findResourceGroupInv(infos []*resource.Info) (*resource.Info, error) {
+	for _, info := range infos {
+		isInv, err := live.IsResourceGroupInventory(info)
+		if err != nil {
+			return nil, err
+		}
+		if isInv {
+			return info, nil
+		}
+	}
+	return nil, fmt.Errorf("resource group inventory object not found")
+}
+
+// NewCmdMigrate returns the cobra command for the migrate command.
+func NewCmdMigrate(f cmdutil.Factory, ioStreams genericclioptions.IOStreams) *cobra.Command {
+	mo := NewMigrateOptions(f, ioStreams)
+	cmd := &cobra.Command{
+		Use:                   "migrate DIRECTORY",
+		DisableFlagsInUseLine: true,
+		Short:                 i18n.T("Migrate inventory from ConfigMap to ResourceGroup custom resource"),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return mo.Run(cmd.InOrStdin(), args)
+		},
+	}
+	cmd.Flags().StringVar(&mo.initOptions.name, "name", "", "Inventory object name")
```

Comment:
I think we are trying avoid this pattern. Can we just us the runner pattern here like we do most other places and then only bind cobra flags to the runner?